### PR TITLE
[DISCO-2514] Merino Test Docs Hub

### PR DESCRIPTION
**Pull request by taddes**
## References

JIRA: [DISCO-2514](https://mozilla-hub.atlassian.net/browse/DISCO-2514)

## Description
- Moved documentation to our central docs hub under `testing/`
- Updated all direct links to docs and files
- Updated some wording to be clearer and more concise
- updated Readme files in individual directories to point to relevant docs
- moved and updated `test-strategy` to be more specific, linked out to functional docs

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-2514]: https://mozilla-hub.atlassian.net/browse/DISCO-2514?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/testing/contract-tests/index.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Contract Tests
```
Comment (mtbottle) [inline docs/testing/contract-tests/index.md:1]: I think we can reduce one page (and reduce one click-through) by having the overview in here instead. So, the content in `contract-tests/contract-tests.md` can be moved here.

**Review by mtbottle — CHANGES_REQUESTED** [review]
A few suggestions and some of the links are not working.

File: docs/testing/testing-strategy.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+[test_dir]: /tests/
+[test_docs_dir]: /docs/testing/
+[unit_tests]: /tests/unit/
+[unit_tests_docs]: /docs/testing/unit-tests.md
+[integration_tests]: /tests/integration/
+[integration_tests_docs]: /docs/testing/integration-tests.md
+[contract_tests]: /tests/contract/
+[contract_tests_docs]: /docs/testing/contract-tests/
+[load_tests]: /tests/load/
+[load_tests_docs]: /docs/testing/load-tests.md
```
[the remark this answers is not in the document]
Comment (taddes) [reply docs/testing/testing-strategy.md:30]: Hmm, what IDE are you using? All these links work fine for me in VSCode and Pycharm.  I was avoiding relative links which look messy and have seen this in our docs, but are you refering to when mdbook creates the files? Does it not support absolute paths?

[the remark this answers is not in the document]
Comment (mtbottle) [reply docs/testing/testing-strategy.md:30]: I run `make doc-preview` and it creates a local `mdbook` server. That's what gets published to https://mozilla-services.github.io/merino-py/ .

**Comment by taddes** [comment]
All changes made @mtbottle with links to GitHub. Just note the links in preview are broken for the docs on `index` just because the PR is not yet merged. Once merged, those links will be active.

**Review by mtbottle — APPROVED** [review]
Just one comment about linking documentation: if we're referencing documentation (not code), let's use relative paths so that we stay in the same documentation system. Feel free to merge this once the links have been fixed.

File: docs/testing/index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+All Contract tests must pass and load tests cannot go below a minimum performance threshold.
+
+Test documentation resides in the [/docs/testing/][test_docs_dir] directory.
+
+The functional test strategy is four-tiered, composed of: 
+
+- [unit][unit_tests] - [documentation][unit_tests_docs]
+- [integration][integration_tests] - [documentation][integration_tests_docs]
+- [contract][contract_tests] - [documentation][contract_tests_docs]
+- [load][load_tests] - [documentation][load_tests_docs]
```
[the remark this answers is not in the document]
Comment (taddes) [reply docs/testing/index.md:20]: Absolutely, good call. Will make the revisions.

**Pull request merged**